packages/swing-store/src/snapStore.js

```diff
@@ -154,72 +189,41 @@ export function makeSnapStore(
           snapReader.destroy();
         });
 
-        // Create a file for the compressed snapshot in-place
-        // to be atomically renamed once we know its name
-        // from the hash of raw snapshot contents.
-        const {
-          path: tmpGzPath,
-          fd: tmpGzFd,
-          cleanup: tmpGzCleanup,
-        } = await ptmpFile({ tmpdir: root });
-        cleanup.push(tmpGzCleanup);
-        const gzWriter = createWriteStream(noPath, {
-          fd: tmpGzFd,
-          autoClose: false,
-        });
-        cleanup.push(() => gzWriter.close());
-
-        await Promise.all([fsStreamReady(snapReader), fsStreamReady(gzWriter)]);
+        await fsStreamReady(snapReader);
 
         const hashStream = createHash('sha256');
         const gzip = createGzip();
+        let compressedByteCount = 0;
 
         const { result: hash, duration: compressSeconds } =
           await measureSeconds(async () => {
             snapReader.pipe(hashStream);
-            snapReader.pipe(gzip).pipe(gzWriter);
-
-            await Promise.all([finished(snapReader), finished(gzWriter)]);
+            const compressedSnapshot = await buffer(snapReader.pipe(gzip));
+            await finished(snapReader);
 
             const h = hashStream.digest('hex');
-            await pfsync(tmpGzFd);
-
-            const tmpGzClose = cleanup.pop();
-            tmpGzClose();
+            sqlStopUsingLastSnapshot.run(vatID);
+            if (!keepSnapshots) {
+              deleteAllUnusedSnapshots();
+            }
+            compressedByteCount = compressedSnapshot.length;
+            sqlSaveSnapshot.run(
+              vatID,
+              endPos,
+              h,
+              rawByteCount,
+              compressedByteCount,
+              compressedSnapshot,
+            );
 
             return h;
           });
 
-        toDelete.delete(hash);
-        const baseName = baseNameFromHash(hash);
-        const filePath = pathResolve(root, baseName);
-        const infoBase = {
-          filePath,
+        return freeze({
```

Comment:
Why `freeze` instead of `harden`? We're using `assert`, which means we're already committed to running under SES. The object is shallow, so the only difference is telling `harden`'s WeakMap that it can ignore the object if it appears again later, but still `harden` seems like the standard to follow.

Reply:
Don't ask me man, I didn't do it.  I think it's just a goof somewhere upstream; I'll change it and I expect nothing will go off the rails.

Reply:
oh, hah, sorry.. thanks for fixing it